managed/services/server/updater.go

```diff
@@ -0,0 +1,220 @@
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/percona/pmm/api/serverpb"
+	"github.com/percona/pmm/version"
+)
+
+// Updater is a service to check for updates and trigger the update process.
+type Updater struct {
+	l                  *logrus.Entry
+	supervisord        supervisordService
+	watchtowerHostname *url.URL
+}
+
+// NewUpdater creates a new Updater service.
+func NewUpdater(supervisord supervisordService, watchtowerHostname *url.URL) *Updater {
+	return &Updater{
+		l:                  logrus.WithField("service", "updater"),
+		supervisord:        supervisord,
+		watchtowerHostname: watchtowerHostname,
+	}
+}
+
+func (s *Updater) sendRequestToWatchtower(ctx context.Context, newImageName string) error {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return errors.Wrap(err, "failed to get hostname")
+	}
+
+	u, err := s.watchtowerHostname.Parse("/v1/update")
+	if err != nil {
+		return errors.Wrap(err, "failed to parse URL")
+	}
+	q := u.Query()
+	q.Set("hostname", hostname)
+	q.Set("newImageName", newImageName)
+	u.RawQuery = q.Encode()
+
+	// Create a new request
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return errors.Wrap(err, "failed to create request")
+	}
+
+	// Add the token to the request header
+	token := os.Getenv("PMM_WATCHTOWER_TOKEN")
+	if token != "" {
+		req.Header.Set("Authorization", "Bearer "+token)
+	}
+
+	// Send the request
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "failed to send request")
+	}
+	defer resp.Body.Close() //nolint:errcheck
+
+	// Check the response
+	if resp.StatusCode != http.StatusOK {
+		return errors.Errorf("received non-OK response: %v", resp.StatusCode)
+	}
+
+	s.l.Info("Successfully triggered update")
+	return nil
+}
+
+func (s *Updater) currentVersion() string {
+	return version.Version
+}
+
+// StartUpdate triggers the update process.
+func (s *Updater) StartUpdate(ctx context.Context, newImageName string) error {
+	if newImageName == "" {
+		latest, err := s.latest(ctx)
+		if err != nil {
+			s.l.WithError(err).Error("Failed to get latest version")
+			newImageName = "perconalab/pmm-server:3-dev-latest"
```

Comment:
Can we make it a const and put it at the very top?

Reply:
good point, will do